### PR TITLE
Fixes #9418 : add notes_for_tags method in user model

### LINKED ISSUE #9418: NoMethodError: undefined method `notes_for_tags' for #<User:0x00007f95d21db590>
Sentry Issue: [PLOTS2-18A](https://sentry.io/organizations/publiclab/issues/2310388476/?referrer=github_integration)

```
NoMethodError: undefined method `notes_for_tags' for #<User:0x00007f95d21db590>
Did you mean?  note_streak
  app/controllers/notes_controller.rb:336:in `author_topic'
    @notes = @user.notes_for_tags(@tagnames)
...
(104 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by toshitapandey**
Fixes https://github.com/publiclab/plots2/issues/9418 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

#### Screenshot
URL - `http://localhost:3000/notes/author/admin/lat:53`
image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -517,6 +517,14 @@ def drafts
     .order('created DESC')
   end
 
+  def notes_for_tags(tagnames)
+    Node.includes(:node_tag, :tag)
+        .where('term_data.name IN (?)', tagnames)
```
Comment (TildaDares) [inline app/models/user.rb:522]: Hi @toshitapandey, can you make this addition? We need to search by the user in the request params. Thanks!
```suggestion
        .where('term_data.name IN (?)', tagnames)
        .where(uid: uid)
```

**Review by TildaDares — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (toshitapandey) [reply app/models/user.rb:522]: @TildaDares - Updated the query. Please take a look.

Comment (toshitapandey) [reply app/models/user.rb:522]: @TildaDares - please take a look.

**Review by TildaDares — APPROVED** [review]
This looks good to me! Let's get a second opinion from @jywarren.

**Comment by jywarren** [comment]
Great job, this looks good!

**Pull request merged**